src/command.rs

```diff
@@ -436,6 +443,11 @@ pub fn run() -> Result<()> {
 			let collator_options = cli.run.collator_options();
 
 			runner.run_node_until_exit(|config| async move {
+				let hwbench = (!cli.no_hardware_benchmarks).then_some(
+					config.database.path().map(|database_path| {
+						let _ = std::fs::create_dir_all(database_path);
```

Comment:
**MAYBE CR**

Usually the db takes care of creating the folders on the path to the db. 
* Why do we need to do that here too?
*  Isn't the client already instantiated at this point? 
* We do not care for the result, as?

Reply:
Nevermind. Client is not there yet. See why we need to create that path here. What a hack... Is this copied from parity?